File: mergesort/mergeSort.py
def mergeSort(arr):
  # Divide
  # Conquer
  # Combine
  
  
  if len(arr) > 1:
    mid = len(arr) // 2
    L = arr[:mid]
    R = arr[mid:]
    
    mergeSort(L)
    mergeSort(R)
    
    i = 0
    j = 0
    k = 0
    
    print("arrBefore: ", arr)
  
    while i < len(L) and j < len(R):
      if L[i] <= R[j]:
        arr[k] = L[i]
        i += 1
      elif L[i] > R[j]:
        arr[k] = R[j]
        j += 1
      k += 1
      
    print("arrAfter: ", arr)
    
    while i < len(L):
      arr[k] = L[i]
      i += 1
      k += 1
    
    while j < len(R):
      arr[k] = R[j]
      j += 1
      k += 1
    


arr = [4,3,5,1,73,5,2,2,1,4,6]
mergeSort(arr)
print(arr)

File: mergesort/mergeSort.js
const mergeSort = (arr) => {
  // Divide
  // Conquer
  // Combine

  if (arr.length > 1) {
    const mid = Math.floor(arr.length / 2)
    const L = arr.slice(0, mid)
    const R = arr.slice(mid)
    
    mergeSort(L)
    mergeSort(R)
    
    console.log(R);
    
    let i = 0, j = 0, k = 0

    console.log(arr)
    
    while (i < L.length && j < R.length) {
      if (L[i] <= R[j]) {
        arr[k] = L[i];
        i++
      } else if (L[i] > R[j]) {
        arr[k] = R[j];
        j++;
      }
      console.log(k)
      k++
    }

    console.log(arr)
    
    while (i < L.length) {
      arr[k] = L[i];
      i++;
      k++
    }
    
    while (j < R.length) {
      arr[k] = R[j];
      j++;
      k++;
    }
    // console.log(L)
    // console.log(R)

  }


}

const arr = [4,3,5,1,7, 2, 6, -2342]
mergeSort(arr)
console.log(arr)